lib/SILGen/SILGenBridging.cpp

```diff
@@ -1050,8 +1050,28 @@ void SILGenFunction::emitNativeToForeignThunk(SILDeclRef thunk) {
       if (attr->isSwift3Inferred() &&
           !decl->getAttrs().hasAttribute<DynamicAttr>() &&
           !getASTContext().LangOpts.isSwiftVersion3()) {
-        B.createBuiltin(loc, getASTContext().getIdentifier("swift3ImplicitObjCEntrypoint"),
-            getModule().Types.getEmptyTupleType(), { }, { });
+        
+        // Get the starting source location of the declaration so we can say
+        // exactly where to stick '@objc'.
+        SourceLoc objcInsertionLoc = decl->getStartLoc();
```

Comment:
you want `getAttributeInsertionLoc(/*forModifier=*/false)` here, not `getStartLoc()`.